### PR TITLE
Synthetic source: don't allow disabling source

### PR DESCRIPTION
**Pull request by nik9000**
This rejects mappings that disable the `_source` but also configure it
to be `synthetic` because that sort of configuration doesn't make sense
`synthetic` is a way to enable source at a low cost.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java [hunk 1]
```diff
@@ -99,6 +99,9 @@ public SourceFieldMapper build() {
                 && excludes.getValue().isEmpty()) {
                 return DEFAULT;
             }
+            if (enabled.getValue() == false && synthetic.getValue()) {
+                throw new IllegalArgumentException("_source may not be disabled when setting [synthetic: false]");
```
[the remark this answers is not in the document]
Comment (nik9000) [reply server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java:103]: Yup.

**Comment by nik9000** [comment]
@romseygeek could you have another look at this one?

**Review by romseygeek — APPROVED** [review]
LGTM

**Comment by nik9000** [comment]
Thanks @romseygeek !

**Pull request merged**